### ISSUE TITLE
Improve MeetEcho access to slides during meetings

### ISSUE THREAD

**Issue by rjsparks** [ietf-tools/datatracker#7005]
MeeEcho needs a pair of specialized APIs to replace their current use of /api/v1 so that chair experience at meetings can be improved.

1) The datatracker should provide an API for the session that lists the slide names,  order, content locations, and titles for a session at the time of the retrieve. This can probably be a public API. 

2) The datatracker should POST to a protected API at MeetEcho whenever slides for a session are changed (new versions uploaded, new slides added, slides deleted, slides reordered). We'll work with MeetEcho out of band to determine if this just triggers a refetch of API 1) or if it provides the new state as part of the POST.

**Comment by jennifer-richards** [comment]
> 1. The datatracker should provide an API for the session that lists the slide names,  order, content locations, and titles for a session at the time of the retrieve. This can probably be a public API.

There's the existing `/api/meeting/session/<session_id>/materials` API that returns things like..
```json
{
  "url": "https://datatracker.ietf.org/meeting/118/materials/agenda-118-moq-01",
  "slides": {
    "decks": [
      {
        "id": 121879,
        "title": "Chair slides",
        "url": "https://datatracker.ietf.org/meeting/118/materials/slides-118-moq-chair-slides",
        "ext": "pdf"
      },
```
Does the `slides.decks` section of that do the right thing? It has the slide title but not the "slide name" - not sure what that would be. Filename? The decks _are_ ordered in the array.

**Comment by jennifer-richards** [comment]
oob, @alexamirante pointed out that the JSON returned from the existing end point does not include the rev and that it'd be convenient to have the order available as a field instead of having to infer it from the position in the list.

I think we can add both to that response.  That API is used by the agenda display, which I think will ignore fields it does not care about. I think it's better to reuse this instead of adding another nearly redundant API endpoint. @NGPixel, any concerns with that?

**Comment by rjsparks** [comment]
Be sure to review the caching of that endpoint - if there is some, there needs to be a way to bypass it for MeetEcho's use case.

For `name` - Meetecho is planning to shift to title I think. But if we need to provide something other than title and url to let chairs find the right thing: the original description notes that 'slides-118-moq-` isn't helping poeple much when they're already looking at just that session. It's worse for `slides-interim-2024-tools-2-`. Maybe we should have a `name` (with a better name) that strips away that prefix from the filename?

**Comment by alexamirante** [comment]
> For `name` - Meetecho is planning to shift to title I think.

Exactly. `title` is what we need. No need to add a further `name` field.

**Comment by alexamirante** [comment]
@jennifer-richards I confirm that the proposed API enriched with `rev` and `order` fields has all info we need.

**Comment by jennifer-richards** [comment]
> Be sure to review the caching of that endpoint - if there is some, there needs to be a way to bypass it for MeetEcho's use case.

We're currently bypassing the `/api` tree so this should not need special accommodation.

I'll proceed without a shortened `name` for now.

**Comment by alexamirante** [comment]
LGTM! Can you please confirm that when you submit a revision the `id` stays the same and only the `rev` changes?

> I noticed we're always sending the versionless href for the slides, which will retrieve the latest rev. I think that's desirable

That's definitely desirable to me!

**Comment by jennifer-richards** [comment]
Yes, the `id` will remain fixed.

Also, just for completeness, the JSON I quoted was the result of piping the API response through `jq .slides.decks`

**Comment by rjsparks** [comment]
released in 12.8.0
